src/devices/Mpu9250/Ak8963Attached.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.Ak8963;
+using System;
+using System.Device;
+using System.Device.I2c;
+
+namespace Iot.Device.Mpu9250
+{
+    internal class Ak8963Attached : Ak8963Interface
+    {
+
+        public override void WriteRegister(I2cDevice i2cDevice, Ak8963.Register reg, byte data)
+        {
+            Span<byte> dataout = stackalloc byte[2] { (byte)Register.I2C_SLV0_ADDR, Ak8963.Ak8963.DefaultI2cAddress };
+            i2cDevice.Write(dataout);
+            dataout[0] = (byte)Register.I2C_SLV0_REG;
+            dataout[1] = (byte)reg;
+            i2cDevice.Write(dataout);
+            dataout[0] = (byte)Register.I2C_SLV0_DO;
+            dataout[1] = data;
+            i2cDevice.Write(dataout);
+            dataout[0] = (byte)Register.I2C_SLV0_CTRL;
+            dataout[1] = 0x81;
+            i2cDevice.Write(dataout);
+        }
+
+        public override byte ReadByte(I2cDevice i2cDevice, Ak8963.Register reg)
+        {
+            Span<byte> read = stackalloc byte[1] { 0 };
+            ReadByteArray(i2cDevice, reg, read);
```

Comment:
doesn't I2cDevice have ReadByte? if not could you create an issue?

Reply:
It has, it's just that I built primitives using ReadByte and ReadByteArray for this sensor. It is easier and faster because, yes, there is a native ReadByte. In this case, beause the sensors needs as many opetation to read 1 or multiple bytes, I just call the main primitive (in this case ReadByteArray)